.github/workflows/osx_build.yml

```diff
@@ -0,0 +1,52 @@
+name: MacOS build and test rolesanywhere-credential-helper
+on:
+  push:
+    branches: [ main ]
+  
+permissions:
+  statuses: write
+  id-token: write
+
+jobs:
+  OSX:
+    name: Build on ${{matrix.runner_image}}
+    runs-on: ${{matrix.runner_image}}
+    strategy:
+      matrix:
+        include:
+          - runner_image: macos-13
+            s3_path: X86_64/OSX/MacOS
+            display_name: "MacOS Intel"
```

Comment:
Minor, but consistency in the format adhered to by the display name might be good to consider. I think the ISA might be better to use as the second word in the display name, so "MacOS x86-64" and "MacOS Aarch64" (I'm using "Aarch64" here since that's the term we use elsewhere generally).